clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -899,6 +907,12 @@ public void unsubscribe() {
     @Override
     public void assign(Collection<TopicPartition> partitions) {
         acquire();
+
+        if (!this.assignment().isEmpty() && this.autoCommitEnabled)
```

Comment:
Both suggestions make sense. I'll update the code to reflect these. `maybeAutoCommitOffsetAsync` commits offsets for `subscriptions.allConsumed()` which should cover the above check `!this.assignment().isEmpty()`. If there nothing has been consumed no commit is made.